new format ivf (VP8): still very simple, no frame disassembly, but it…

frankpolte opened the pull request:
… is a start. works with the ivf files from https://github.com/webmproject/vp8-test-vectors

GreyCat commented on media/ivf.ksy:
```diff
@@ -0,0 +1,67 @@
+    doc: This should be 0
+  - id: headerlen
+    type: u2
+    doc: Normally the header length is 32 byte
```
Could you make this `len_header` to match [style guide](https://doc.kaitai.io/ksy_style_guide.html#attr-id) recommendation?

GreyCat commented on media/ivf.ksy:
```diff
@@ -0,0 +1,67 @@
+  id: ivf
+  file-extension: ivf
+  endian: le
+  title: VP8 raw file format
```
Please remove word "format" from the title — just "VP8 raw file" will work, as in certain contexts this "title" would be used as `$title format`.

GreyCat commented on media/ivf.ksy:
```diff
@@ -0,0 +1,67 @@
+    doc: the timescale is a divider of the seconds (VPX is integer math only) mostly 1000
+  - id: framecount
+    type: u4
+    doc: the number of frames (if not a camera stream)
```
`num_frames`?

GreyCat commented on media/ivf.ksy:
```diff
@@ -0,0 +1,67 @@
+meta:
+  id: ivf
```
Formats repo have global IDs without any namespacing. Given that "ivf" is not a very popular format and there might be other formats identified as "ivf", may I suggest to rename this (and the file) to something more verbose, like `vp8_ivf`, as this is clearly a container for VP8?

GreyCat commented on media/ivf.ksy:
```diff
@@ -0,0 +1,67 @@
+        #size-eos: true
+  block:
+    seq:
+      - id: framesize
```
`len_frame`?

frankpolte commented:
@GreyCat : I renamed and cleaned up the stuff. Even the file was renamed, so that we can add other codec ivf files e.g. vp9_ivf and so on. Right now it is for all codecs, but I plan for parsing vp8 specifically. Is there a way to reference the outer stuff from a more global ksy file?
Maybe a `ivf_mediacontainer.ksy` holds the main header stuff and `vp8_ivf.ksy`, `vp9_ivf.ksy` or `xyz_ivf.ksy` only parse the frame content but including  `ivf_mediacontainer.ksy`.

KOLANICH commented on media/vp8_ivf.ksy:
```diff
@@ -0,0 +1,64 @@
+  license: CC0-1.0
+doc-ref: https://wiki.multimedia.cx/index.php/IVF
+seq:
+## header start
```
I usually prefer the code be self-documenting. Should you remove these comments and use subtypes instead?

KOLANICH commented on media/vp8_ivf.ksy:
```diff
@@ -0,0 +1,64 @@
+  - id: len_header
+    type: u2
+    doc: Normally the header length is 32 byte
+  - id: codec
```
I don't think it is meant to be hardcoded.

GreyCat commented:
> Is there a way to reference the outer stuff from a more global ksy file?

Sure, we have  [imports mechanism](http://doc.kaitai.io/user_guide.html#_importing_types_from_other_files) for that.

I guess everything looks good now, so I'm merging it. Thanks again!
